src/lib/node/message_handler.ts

```diff
@@ -42,6 +49,27 @@ export class NodeMessageHandler {
           node.getValueMetadata(message.valueId),
           client.schemaVersion
         );
+      case NodeCommand.beginFirmwareUpdateGuessFormat:
+        const format = guessFirmwareFileFormat(
+          message.firmwareFilename,
+          message.firmwareFile
+        );
+        actualFirmware = extractFirmware(message.firmwareFile, format);
+        await node.beginFirmwareUpdate(
+          actualFirmware.data,
+          actualFirmware.firmwareTarget
+        );
+        return {};
+      case NodeCommand.beginFirmwareUpdateKnownFormat:
+        actualFirmware = extractFirmware(
+          message.firmwareFile,
```

Comment:
are you sure about that? We are receiving buffers in the library today, albeit very small ones.

Re: file sizes, I was told up to the hundreds of KBs. I have two firmware files on hand that are between 120-160 KB. When they are JSON serialized though they are ~3.5x the size.

Sending the raw binary is probably the way to go but that does increase the complexity of the solution significantly because the way the server is currently written, it always expects JSON

Reply:
the buffers that we receive are processed to a special format like `{"type": "_buffer", "data": "..."}`. That's not official JSON.